### PR TITLE
#80 Added an `onboarding_controller` for onboarding a first time user.

### PR DESCRIPTION
**Pull request by AndrewMcBurney**
### Overview

#80 FastlaneCI can now run without having the needed environment variables or proper configuration setup without failing.

### Changes

Upon launching the server without the proper configuration, the user will be directed to the `/onboarding` page. They will not be able to access other routes until they complete the onboarding.

Once the onboarding is finished, the user will not be able to access the onboarding page (they will be redirected to `/`).

Should the configuration values be modified at any time causing an invalid state (i.e, needed environment variables removed), or the remote configuration repo deleted, they will be directed back to the onboarding page to fix their invalid state.

screen shot 2018-03-10 at 5 04 21 pm

### User Experience

screen shot 2018-03-10 at 5 04 27 pm

> The onboarding page will tell the user what configuration values they are missing (environment variables, remote or local configuration repository).

___

screen shot 2018-03-10 at 5 04 44 pm

___

screen shot 2018-03-10 at 5 04 53 pm

___

screen shot 2018-03-10 at 5 05 11 pm

___

screen shot 2018-03-10 at 5 05 22 pm

___

screen shot 2018-03-10 at 5 05 35 pm

___

screen shot 2018-03-10 at 5 05 42 pm

### CODE REVIEW

**Comment by minuscorp** [comment]
Relates #253 should I abandon that PR on favor of this? I think the only thing this don't have is the addition of a manual trigger by default. If you want you can pick my changes and add them in this PR

File: features/onboarding/views/forms/_ci_bot_account.erb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<!-- TODO: setup bot permisisons for the repo, etc. -->
+
+<form id="ci_bot_account" method="POST" action="/onboarding/ci_bot_account">
+  <h4>2. CI Bot Account</h4>
+
+  <p>
+    <code>fastlane.ci</code> requires you setup a CI <i>bot account</i> to
```
Comment (AndrewMcBurney) [inline features/onboarding/views/forms/_ci_bot_account.erb]: Any help with this wording would be greatly appreciated. LOL :joy:

File: services/onboarding_service.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  # Provides operations to create and mutate the FastlaneCI configuration
+  # repository
+  class OnboardingService
+    include FastlaneCI::Logging
+
+    # Spawns a process that opens the onboarding page using the user's preferred
+    # web browser
+    def launch_onboarding!
+      logger.info("Launching onboarding web page.")
+      `open http://localhost:8080/onboarding`
```
Comment (AndrewMcBurney) [inline services/onboarding_service.rb]: TODO: I need to change this url. 😳

File: features/onboarding/views/forms/_ci_bot_account.erb (hunk 1 above)
Comment (minuscorp) [reply features/onboarding/views/forms/_ci_bot_account.erb]: I'd really use something like [this](http://danger.systems/guides/getting_started.html#creating-a-bot-account-for-danger-to-use)

Comment (AndrewMcBurney) [reply features/onboarding/views/forms/_ci_bot_account.erb]: Thank you!! I'll update the documentation to be more like this page.

File: services/onboarding_service.rb (hunk 2 above)
Comment (minuscorp) [reply services/onboarding_service.rb]: Something that Jenkins do is to provide in the logs (for the first time) a link url with the password hash in order to authenticate the first user. Something like:
```
_fastlane.ci_ has launched successfully, visit URL in order to finish the onboarding process
```
Also, I'll love to see in the page that is opened:
`I have an existing ci-repo`, `I don't have an existing ci-repo`

**Comment by taquitos** [comment]
Omg these screens look amazing.

**Pull request merged**